.azure-devops/templates/run-tests-parallel.yml

```diff
@@ -93,6 +85,30 @@ steps:
           inlineScript: |
             export COVERAGE_FILE=.coverage.${{ parameters.name }}
             pytest -vv ${{ parameters.path }} -k "_int.py" --dist=loadfile -n ${{ parameters.num_threads }} --reruns ${{ parameters.num_reruns }} --reruns-delay ${{ parameters.reruns_delay }} --cov=azext_iot --cov-config .coveragerc --junitxml=junit/test-iotext-int.xml --deselect=azext_iot/tests/iothub/core/test_iothub_discovery_int.py::TestIoTHubDiscovery::test_iothub_targets --durations=0
+        ${{ if contains(parameters.path, 'azext_iot/tests/dps') }}:
+          inlineScript: |
+            IOTHUB=`echo test-dps-hub-py$RANDOM`
+            sudo apt install dos2unix -y -q
+            dos2unix pytest.ini
+            CREATE_HUB=`grep -c "azext_iot_testdps_hub=$" pytest.ini`
+
+            if (($CREATE_HUB > 0 && (${{ parameters.azureCLIVersion }} == "released" || ${{ parameters.azureCLIVersion }} == "min")))
+            then
+              echo $IOTHUB
+              az iot hub create -n $IOTHUB -g $(AZEXT_IOT_TESTRG)
+              sed -z -i 's/azext_iot_testdps_hub=\n/azext_iot_testdps_hub='"$IOTHUB"'\n/g' pytest.ini
+              cat pytest.ini
+            fi
+
+            export COVERAGE_FILE=.coverage.${{ parameters.name }}
+            pytest -vv ${{ parameters.path }} -k "_int.py" --dist=loadfile -n ${{ parameters.num_threads }} --reruns ${{ parameters.num_reruns }} --reruns-delay ${{ parameters.reruns_delay }} --cov=azext_iot --cov-config .coveragerc --junitxml=junit/test-iotext-int.xml --durations=0
+            e=$?
+
+            if (($CREATE_HUB > 0))
```

Comment:
This condition also needs to be updated to include the CLI version check

Reply:
a delete doesnt error if the hub does not exist (its a no op, which is what services are supposed to do) so will leave this as is.

Reply:
changed